src/util/native_fs_utils.js

```diff
@@ -580,6 +588,25 @@ function get_bucket_tmpdir_full_path(bucket_path, bucket_id) {
     return path.join(bucket_path, get_bucket_tmpdir_name(bucket_id));
 }
 
+
+/**
+ * translate_error_codes we translate FS error codes to rpc_codes (strings)
+ * and add the rpc_code property to the original error object
+ * @param {object} err
+ * @param {('OBJECT'|'BUCKET'|'USER'|'ACCESS_KEY')} entity
+ */
+function translate_error_codes(err, entity) {
+    if (err.rpc_code) return err;
+    if (err.code === 'ENOENT' || err.code === 'ENOTDIR') {
```

Comment:
@romayalon I just want to make sure that you noticed that the error code `ENOTDIR` will be mapped to `NO_SUCH_${entity}`, and not only 'NO_SUCH_OBJECT' (as was added in PR #7995)